### PR TITLE
feat: better failure when failing promise

### PR DESCRIPTION
**Pull request by alacambre-yseop**
see https://stackoverflow.com/a/45858343/3577898

### CODE REVIEW

File: client/test/extension.test.ts [hunk 1]
```diff
@@ -14,43 +14,52 @@ import * as vscode from 'vscode';
...
-    test("Test extension adds YML support.", (done) => {
-        vscode.languages.getLanguages().then((languages) => {
+    test("YML extension adds YML support.", (done) => {
+        var ymlExtension = vscode.extensions.all.find(extension => {
+            return extension.id === "Yseop.vscode-yseopml";
+        });
+        if(!ymlExtension) {
+            assert.fail("Extension doesn't exist.")
+        }
+        vscode.languages.getLanguages()
```
Comment (mminot-yseop) [inline client/test/extension.test.ts]: I take it you added stuff to make tests more independent?

File: client/test/extension.test.ts [hunk 2]
```diff
@@ -14,43 +14,52 @@ import * as vscode from 'vscode';
...
     });
     
-    test("Test extension adds YML support.", (done) => {
-        vscode.languages.getLanguages().then((languages) => {
+    test("YML extension adds YML support.", (done) => {
+        var ymlExtension = vscode.extensions.all.find(extension => {
+            return extension.id === "Yseop.vscode-yseopml";
+        });
+        if(!ymlExtension) {
+            assert.fail("Extension doesn't exist.")
```
Comment (mminot-yseop) [inline client/test/extension.test.ts]: Detail but still: `if(bool) assert fail` looks weird. Isn't there an `assertTrue` or something?

File: client/test/extension.test.ts [hunk 3]
```diff
@@ -14,43 +14,50 @@ import * as vscode from 'vscode';
...
     });
 
-    test("Test that not activated YML extension is not active.", () => {
+    test("not activated YML extension is not active.", () => {
         var ymlExtension = vscode.extensions.all.find(extension => {
             return extension.id === "Yseop.vscode-yseopml";
         });
-        assert.notEqual(null, ymlExtension);
-        assert.equal(false, ymlExtension.isActive);
+        assert.notStrictEqual(ymlExtension, null);
```
[the remark this answers is not in the document]
Comment (alacambre-yseop) [reply client/test/extension.test.ts]: `notEqual` is deprecated and `notStrictEqual` seems to be the method to use.

[the remark this answers is not in the document]
Comment (mminot-yseop) [reply client/test/extension.test.ts]: wow

**Pull request merged**